src/Agent.Worker/DiagnosticLogManager.cs

```diff
@@ -0,0 +1,247 @@
+
+            // upload it
+            jobServerQueue.QueueFileUpload(
+                    timelineId: executionContext.MainTimelineId, 
+                    timelineRecordId: executionContext.TimelineId, 
+                    // type: CoreAttachmentType.DiagnosticLog, may need to rev dependency version?
+                    type: "DistributedTask.Core.DiagnosticLog", 
+                    name: metadataFileName, 
+                    path: metadataFilePath, 
+                    deleteSource: false);
+
+            // upload the diagnostics zip file
+            executionContext.Debug("Uploading Diagnostic zip file.");
+            jobServerQueue.QueueFileUpload(
+                    timelineId: executionContext.MainTimelineId, 
+                    timelineRecordId: executionContext.TimelineId, 
+                    // type: CoreAttachmentType.DiagnosticLog,
+                    type: "DistributedTask.Core.DiagnosticLog", 
+                    name: diagnosticsZipFileName, 
+                    path: diagnosticsZipFilePath, 
+                    deleteSource: false);
+
+            // Delete support folder
+            // TODO: We can't delete here. The file upload is queued so they will be gone when its time to upload. Will normal cleanup take care of it?
+
+            executionContext.Debug("Diagnostic file upload complete.");
+        }
+
+        // The current solution is a hack. We need to rethink this and find a better one.
+        // The list of worker log files isnt available from the logger. It's also nested several levels deep.
+        // For this solution we deduce the applicable worker log files by comparing their create time to the start time of the job.
+        private List<string> GetWorkerDiagLogFiles(string diagFolder, DateTime jobStartTimeUtc)
+        {
+            // Get all worker log files with a timestamp equal or greater than the start of the job
+            var workerLogFiles = new List<string>();
+            var directoryInfo = new DirectoryInfo(diagFolder);
+            foreach (FileInfo file in directoryInfo.GetFiles().Where(f => f.Name.StartsWith("Worker_")))
+            {
+                // The format of the logs is:
+                // Worker_20171003-143110-utc.log
+                DateTime fileCreateTime = DateTime.ParseExact(s: file.Name.Substring(startIndex: 7, length: 15), format: "yyyyMMdd-HHmmss", provider: CultureInfo.InvariantCulture);
+
+                if (fileCreateTime >= jobStartTimeUtc)
+                {
+                    workerLogFiles.Add(file.FullName);
+                }
+            }
+            
+            return workerLogFiles;
+        }
+
+        private string GetEnvironmentContent(int agentId, string agentName, ReadOnlyCollection<TaskInstance> tasks)
+        {
+            var builder = new StringBuilder();
+
+            builder.AppendLine($"Environment file created at(UTC): {DateTime.UtcNow}"); // TODO: Format this like we do in other places.
```

Comment:
most of the information already exists in the worker log, i think.

Reply:
I will compare each piece of data and see what I can find. It's also easier to quickly check the environment file than have to dig through the worker log?

@bryanmacfarlane thoughts?